### ISSUE TITLE
[HOLD for payment 2024-08-26] [Upgrades] [$250] Workspace - Incorrect app navigation after clicking on back button on upgrade RHP

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#45682]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.9-1
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Go to staging.new.expensify.com
2. Go to More features.
3. Scroll down.
4. Toggle on Report fields.
5. Click on the back button on RHP.
6. Note that the page scrolls up.
7. Enable Accounting.
8. Go to Accounting.
9. Toggle on NetSuite.
10. Click on the back button on RHP.
11. Note that app returns to workspace profile page.

## Expected Result:
In Step 6, page will not scroll up after clicking on back button on RHP
In Step 11, app will remain on the same accounting page after clicking on back button on RHP

## Actual Result:
In Step 6, page scrolls up after clicking on back button on RHP
In Step 11, app remains on the same accounting page after clicking on back button on RHP

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] #45930

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/user-attachments/assets/b59c6029-3a2b-4640-a6ca-233c1ec8154c

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~015d4828d1746446dd</li>
        <li>Upwork Job ID: 1813956464241732548</li>
        <li>Last Price Increase: 2024-07-18</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @garrettmknight</details>

**Comment by roryabraham** [comment]
not 100% sure this needs to block deploy. It's definitely not a back-end issue though

**Comment by lanitochka17** [comment]
The same issue is observed when performing other steps:
1. Create a workspace
2. Navigate to workspace settings page
3. Click on More features > toggle Report fields
4. Click on 'Upgrade' > click on 'Got it, thank'

https://github.com/user-attachments/assets/2805d11c-4c4d-41e4-8342-99a71302ec6d

**Comment by trjExpensify** [comment]
@allroundexperts @garrettmknight @rushatgabhane pretty sure GK just created an issue for this: https://github.com/Expensify/App/issues/45692

**Comment by garrettmknight** [comment]
Yeah, this is a dupe and also not a blocker. Closing in favor of the other one.

**Comment by garrettmknight** [comment]
Sorry @cretadn22 - I think I closed the wrong issue. Reopening here, but @allroundexperts might already be fixing this one in a separate PR.

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
In Step 6, page scrolls up after clicking on back button on RHP
In Step 11, app remains on the same accounting page after clicking on back button on RHP

### What is the root cause of that problem?
When we open the upgrade page from Accounting, we don't pass the `backTo` param to the route, and then we fallback to the policy profile page.

https://github.com/Expensify/App/blob/eec4d6ad371f5c8d89f3d3eddbdc8184c4f7c9a8/src/pages/workspace/upgrade/WorkspaceUpgradePage.tsx#L47
### What changes do you think we should make in order to solve the problem?
We should pass the `backTo` param for all places we navigate to this page as in `WorkspaceMoreFeaturesPage` [here](https://github.com/Expensify/App/blob/eec4d6ad371f5c8d89f3d3eddbdc8184c4f7c9a8/src/pages/workspace/WorkspaceMoreFeaturesPage.tsx#L210). We can use `Navigation.getActiveRouteWithoutParams()` to get the active route

https://github.com/Expensify/App/blob/eec4d6ad371f5c8d89f3d3eddbdc8184c4f7c9a8/src/components/ConnectToNetSuiteButton/index.tsx#L59

https://github.com/Expensify/App/blob/eec4d6ad371f5c8d89f3d3eddbdc8184c4f7c9a8/src/components/ConnectToSageIntacctButton/index.tsx#L66

https://github.com/Expensify/App/blob/eec4d6ad371f5c8d89f3d3eddbdc8184c4f7c9a8/src/pages/workspace/categories/CategorySettingsPage.tsx#L145

https://github.com/Expensify/App/blob/eec4d6ad371f5c8d89f3d3eddbdc8184c4f7c9a8/src/pages/workspace/categories/CategorySettingsPage.tsx#L159

https://github.com/Expensify/App/blob/eec4d6ad371f5c8d89f3d3eddbdc8184c4f7c9a8/src/pages/workspace/tags/TagSettingsPage.tsx#L75


### What alternative solutions did you explore? (Optional)

**Comment by garrettmknight** [comment]
Still confirming with @allroundexperts whether they have a fix in the works. If they don't I'll re-add `Help Wanted`.

**Comment by allroundexperts** [comment]
My PR would fix this issue.

**Comment by garrettmknight** [comment]
Also taking this one off the payment hold while we wait for the updated PR to ship.

**Comment by garrettmknight** [comment]
Going to keep this one open to confirm payment with @rushatgabhane for his review. [Slack convo here.](https://expensify.slack.com/archives/C02NK2DQWUX/p1724855729417049)

**Comment by garrettmknight** [comment]
**Payment summary:**
- Reviewer: @rushatgabhane $250 via NewDot

**Comment by garrettmknight** [comment]
Closing - still waiting on @rushatgabhane to request payment though.

**Comment by JmillsExpensify** [comment]
Payment summary:

 - Contributor: @allroundexperts $250 
 - Reviewer: @rushatgabhane $250
